### PR TITLE
Updated several typos

### PR DESCRIPTION
**Pull request by saahilthukral**
Issue #2 

Have updated several typos. I will work on fixing other grammatical errors soon

### CODE REVIEW

**Review by saahilthukral — COMMENTED** [review]
Updated several typos

File: docs/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "grammarly.selectors": [
```
Comment (jugshaurya) [inline docs/.vscode/settings.json:2]: what will this do? @saahilthukral

File: docs/0.intro.md [hunk 2]
```diff
@@ -4,34 +4,36 @@ sidebar_position: 0
...
 
 <h2 align="center">Hi 👋, I'm Shaurya Singhal 🇮🇳</h2>
-<h4 align="center">🤝 Welcome to Shaurya Notes/Blogs</h4>
+<h4 align="center">🤝 Welcome to Shaurya's Notes/Blogs</h4>
 <p align="center"> <img src="https://komarev.com/ghpvc/?username=jugshaurya&label=Profile Views&color=blue&style=plastic" alt="jugshaurya" /> </p>
 
 ### Experience/Achievements/Recent Failures
 
-- I am a **M.C.A 2rd Year Computer Science Student** @ University of Delhi, Department of Computer Science.
+- I am an **M.C.A 2<sup>nd</sup> Year Computer Science Student** @ the University of Delhi, Department of Computer Science.
```
Comment (jugshaurya) [inline docs/0.intro.md]: leave an as `a` and make t in `the` capital

File: docs/0.intro.md [hunk 3]
```diff
@@ -4,34 +4,36 @@ sidebar_position: 0
...
-- 🔭 I’m currently working on _C++, DataStructures & Algorithm, developing skills, javascript, side projects, open source_.
-- 🌱 I’m currently learning anything _related to Javascript, GraphQL, React, Competitive Programming, Problem Solving._
+- 🔭 I’m currently working on _C++, Data Structures & Algorithms, developing skills, JavaScript, side projects, and open-source_.
+- 🌱 I’m currently learning anything related to _Javascript, GraphQL, React, Competitive Programming, and Problem Solving._
 - 🤔 Anyone looking for help can reach out.
-- 📫 How to reach me: Email me at - **shauryasinghal84@gmail.com**
-- 📫 Visit my Portfolio - **https://www.jugshaurya.in/**
-- ⚡ Fun fact: _I love JavaScript ❤️ but currently in relationship with C++._
+- 📫 How to contact me: Feel free to email me at: **shauryasinghal84@gmail.com**
+- 📫 Visit my Portfolio - [https://www.jugshaurya.in/](https://www.jugshaurya.in/)
```
Comment (jugshaurya) [inline docs/0.intro.md:32]: Nice one.

File: docs/1.stl.md [hunk 4]
```diff
@@ -13,19 +13,19 @@ tags: [STL, CPP, Algo, iterators]
 
   - **Container Adapter/ Derived Containers**
 
-    - Don't support iterators, hence to print the content, use while container is empty thing.
+    - Don't support iterators, hence to print the content, use it while the container is an empty thing.
```
Comment (jugshaurya) [inline docs/1.stl.md]: no leave it as it is! 

infact change it to 
`use while(!container.empty()) thing`.

File: docs/1.stl.md [hunk 5]
```diff
@@ -13,19 +13,19 @@ tags: [STL, CPP, Algo, iterators]
...
     - (build upon simple containers)
-    - (`stack`,`queue`, `deque`, `heap`, `priority queue`)
+    - (`stack`, `queue`, `deque`, `heap`, `priority queue`)
 
   - **Associative Containers**
-    - implement sorted Data structure; data can be searched in O(logn)
-    - `set`, `map`, `multimap`, `multiset`
-    - ordered map/set have an order and are implemented as self balancing BST(Red black Trees).
-    - Keys may not need be integral. 
+    - Implement a sorted Data structure; data can be searched in O(logn)
```
Comment (jugshaurya) [inline docs/1.stl.md]: make it `Implemented as a`

File: docs/1.stl.md [hunk 6]
```diff
@@ -663,13 +668,14 @@ forward_list<int> fl; // singly linked list
...
+- like use lower_bound of set on set(set.lower_bound(key))
+- rather than using lower_bound(set.begin(), set.end(), key) as set does not have random iterator.
 
 ### Thank you
 
-- 🎉 You made it here. I believed in you.
-- You are Awesome and a STL Pro. 🫂
\ No newline at end of file
+- 🎉 You've made it here. I believe in you.
+- You are awesome and an STL Pro. 🫂
```
Comment (jugshaurya) [inline docs/1.stl.md]: make it `believed` not believe and `a` not an.

File: docs/2.cpp.md [hunk 7]
```diff
@@ -417,8 +417,8 @@ int main() {
 ## struct vs class
 
 - Use struct to bundle multiple items together.
-- Everything is `public by default in struct` and `private by default in class`.Else everything is the same!
-- when in a hurry use struct like pair(why? because, the default public).
+- Everything is `public by default in struct` and `private by default in class`. Else everything is the same!
+- when in a hurry use struct-like pair(why? because, the default public).
```
Comment (jugshaurya) [inline docs/2.cpp.md]: make it
`use struct or pair`

File: docs/2.cpp.md [hunk 8]
```diff
@@ -629,13 +630,16 @@ while(getline(is, token, ' ')){
 
 ## Byte by Byte
 
-- The #1 reason people fail interviews is lack of confidence!
-  ● Nervousness == Lack of confidence
-  ● Forgetting stuff == Lack of confidence
-  ● Memorizing == Lack of confidence
+- The #1 reason why people fail interviews is because of a lack of confidence!
```
Comment (jugshaurya) [inline docs/2.cpp.md]: remove `because of a `

File: docs/4.graphs.md [hunk 9]
```diff
@@ -45,22 +45,24 @@ tags: [CPP, Graphs, DSU, Prims,Kruskals, Dijkstra]
...
+  
   ```
           Walk
           /  \
       Path   Trial (<--subcategories)
   ```
+
 - **Eulerian path**
-  - a path that covers every and all **e**dges exactly once. (e and E matches). 
+  - a path that covers every **edge** exactly once. (e and E matches).
```
Comment (jugshaurya) [inline docs/4.graphs.md]: leave it as it is. undo the changes

File: docs/6.gameTheory.md [hunk 10]
```diff
@@ -61,15 +64,17 @@ tags: [CompetitiveProgramming, GameTheory]
...
+- A terminal position is one from which no moves are possible. Then one of the players is declared the winner and the other the loser.
+
+### Properties
+
+- All terminal positions are losing in normal game and winning in misere game (above n=0 was terminal position.)
+- If a player can move to a losing position then he is in a winning position.
+- If a player can move only to the winning positions then he is in a losing position.
 
-#### More Example of Subtraction Games
+### Examples of Subtraction Games
```
Comment (jugshaurya) [inline docs/6.gameTheory.md]: leave it `More Example`

File: docs/6.gameTheory.md [hunk 11]
```diff
@@ -150,18 +157,17 @@ tags: [CompetitiveProgramming, GameTheory]
 
 - This helps in identifying positions in the game with vertices of the graph and moves of the game with edges of the graph.
 - Then we will define a function known as the Sprague-Grundy function that contains more information than just knowing whether a position is a P-position or an N-position.
-- Graphs will always be undirected and infact the DAG.
-
+- Graphs will always be undirected and DAG.
```
Comment (jugshaurya) [inline docs/6.gameTheory.md]: leave it unchanged

File: docs/7.dp.md [hunk 12]
```diff
@@ -6,21 +6,22 @@ tags: [DP]
...
 ---
 
-## Dynamic Programming
+## Dynamic Programming (DP)
 
-- All about remembering the Past.
+- All about remembering the past.
 - `Optimal Substructure`
-  - Can express bigger problem in small sub-problems of same type (Recurrence Relation).
+  - Can express bigger problems in small sub-problems of the same type (Recurrence Relation).
```
Comment (jugshaurya) [inline docs/7.dp.md]: leave it  as `problem`

File: docs/7.dp.md [hunk 13]
```diff
@@ -61,15 +61,16 @@ function rec (<form state(s)>, <constraint state(s)>){
 
 - Note:
   - Time complexity of DP problems = `#states * (1 + Average cost of transitions)`
-- Transitions means number of recursive calls.
+- Transitions mean number of recursive calls.
 - Cost of Transition = extra time taken besides recusive calls.
-- #states is basically dp array size.
+- #states are the DP array size.
```
Comment (jugshaurya) [inline docs/7.dp.md]: undo this change

File: docs/7.dp.md [hunk 14]
```diff
@@ -61,15 +61,16 @@ function rec (<form state(s)>, <constraint state(s)>){
 
 - Note:
   - Time complexity of DP problems = `#states * (1 + Average cost of transitions)`
-- Transitions means number of recursive calls.
+- Transitions mean number of recursive calls.
```
Comment (jugshaurya) [inline docs/7.dp.md]: undo this change

File: docs/7.dp.md [hunk 15]
```diff
@@ -115,12 +117,12 @@ function rec (<form state(s)>, <constraint state(s)>){
 
 - LIS
   - n<sup>2</sup>
-  - nlogn approach [** To be Done]
+  - nlog(n) approach [** To be Done]
```
Comment (jugshaurya) [inline docs/7.dp.md]: no need for this change.

File: docs/9.questions.md [hunk 16]
```diff
@@ -182,13 +181,13 @@ https://www.youtube.com/watch?v=YTtpfjGlH2M&list=PLgUwDviBIf0rGEWe64KWas0Nryn7SC
...
-- [ ] Operating System + Interview Questinos
+- [ ] Operating System + Interview Questions
 - [ ] Networking + Interview Questions
 
-- [ ] OOPS Design + INterview Questions
+- [ ] OOPS Design + Interview Questions
 - [ ] System Dessign + INnterview Questions
 - [ ] SQL vs NOSQL + Project revison
-- [ ] Resume Finess
+- [ ] Resume Fitness
```
Comment (jugshaurya) [inline docs/9.questions.md]: it is `Finess` actually.
undo the change

**Review by jugshaurya — CHANGES_REQUESTED** [review]
Please make changes as mentioned.

**Comment by jugshaurya** [comment]
Also don't forget to star the repository. 😸

File: docs/.vscode/settings.json (hunk 1 above)
Comment (saahilthukral) [reply docs/.vscode/settings.json:2]: This is just a plugin for grammarly vscode. I pushed this file by accident

File: docs/0.intro.md (hunk 2 above)
Comment (saahilthukral) [reply docs/0.intro.md]: Will do

**Comment by saahilthukral** [comment]
Hello I have fixed the new typos according to how you wanted them to be. I hope this helps you

**Review by saahilthukral — COMMENTED** [review]
Updated documents again, pending author's approval

**Comment by jugshaurya** [comment]
Good Work. Thank you.  Appreciated. 🥇 :
